pkgs/by-name/bu/bun/hooks/default.nix

```
@@ -0,0 +1,86 @@
+{
+  lib,
+  stdenvNoCC,
+  jq,
+  moreutils,
+  cacert,
+  makeSetupHook,
+  bun,
+}:
+
+{
+  fetchDeps =
+    {
+      hash ? "",
+      pname,
+      workspaces ? [ ],
+      preInstall ? "",
+      installFlags ? [ ],
+      ...
+    }@args:
+    let
+      args' = builtins.removeAttrs args [
+        "hash"
+        "pname"
+      ];
+    in
+    stdenvNoCC.mkDerivation (
+      finalAttrs:
+      (
+        args'
+        // {
+          name = "${pname}-bun-deps";
+
+          nativeBuildInputs = [
+            cacert
+            jq
+            moreutils
+            bun
+          ] ++ args'.nativeBuildInputs;
+
+          installPhase = ''
+            runHook preInstall
+
+            export HOME=$(mktemp -d)
+            export BUN_INSTALL_CACHE_DIR=$out
+
+            ${preInstall}
+
+            bun install \
+                --force \
+                ${lib.escapeShellArgs (lib.map (package: "--filter=${package}") workspaces)} \
+                ${lib.escapeShellArgs installFlags} \
+                --frozen-lockfile
+
+            runHook postInstall
+          '';
+
+          fixupPhase = ''
+            runHook preFixup
+
+            # Remove timestamp and sort the json files
+            #   rm -rf $out/v3/tmp
+            #   for f in $(find $out -name "*.json"); do
+            #     jq --sort-keys "del(.. | .checkedAt?)" $f | sponge $f
+            #   done
+
+              runHook postFixup
+          '';
```

```suggestion
          # Build a reproducible tarball, per instructions at https://reproducible-builds.org/docs/archives/
          fixupPhase = args.fixupPhase or ''
            runHook preFixup

            tar --sort-name \
              --mtime="@$SOURCE_DATE_EPOCH" \
              --owner=0 --group=0 --numeric-owner \
              --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime \
              -czf $out $BUN_INSTALL_CACHE_DIR

              runHook postFixup
          '';
```
This is something that should've (and still should be but hash invalidation sucks) been in the pnpm hook. We can save *massive* amounts of network traffic and storage by compressing our modules like this, similar to `fetchCargoTarball`

And not sure if what's commented above is actually needed here...? But if it is, just add it above this